lib/experimental/class-wp-nav-menu-conversion-gutenberg.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * WP_Nav_Menu_Conversion_Gutenberg class
+ *
+ * @package gutenberg
+ * @since 6.3.0
+ */
+
+/**
+ * Manages Fallback behavior for Navigation menus.
+ *
+ * @access public
+ */
+class WP_Nav_Menu_Conversion_Gutenberg {
+
+	/**
+	 * The Menu term object of the menu to convert.
+	 *
+	 * @var WP_Term
+	 */
+	private $menu;
+
+	/**
+	 * Constructor
+	 *
+	 * @param WP_Term $menu The Menu term object of the menu to convert.
+	 */
+	public function __construct( $menu ) {
```

Comment:
Yes that's a good point.

It does seem this class is effectively encapsulating a one time operation to convert classic menu X to block menu Y.

No internal "state" (properties) is maintained or operated upon except in the one `convert` function. That could just as easily be an argument passed to the `convert()` method.

Moving to static, we'd still get all the benefits of privacy without the additional overhead of the object which seems unnecessary.

Curious to hear if @hellofromtonya has any thoughts on this?

Reply:
>Moving to static, we'd still get all the benefits of privacy without the additional overhead of the object which seems unnecessary.
>
>Curious to hear if @hellofromtonya has any thoughts on this?

tl;dr
I agree.

Reasoning:

In the current class' design and usage, carrying the original `WP_Term` menu object (the one to be converted) seems unnecessary. As mentioned by @draganescu, its sole usage is for the `convert()` functionality. Once `convert()` happens, that property is no longer needed or used, but continues to consume memory. And as @draganescu and @getdave point out, the conversion is one-time event.

Thinking more long-term as to how this converter class could evolve, the `WP_Nav_Conversion_Gutenberg::convert()` functionality could serve to convert any menu that is injected into it. If that could be a design possibility, then it makes sense to me that this method have the `$menu` parameter in its signature. That shifts its intent from a one-time binding to a single `WP_Term` menu object to any menu that needs to be converted without having to instantiate `WP_Nav_Menu_Conversion_Gutenberg` for each menu. Hmm, this could also reduce processing (instantiating multiple converter objects) and memory (multiple objects in memory).

In essence, this class' purpose is to be a converter utility, rather than individual convert objects with a one-to-one binding to a single menu.

Switching its design from OOP to a class wrapper (static methods) and injecting the menu into the public `convert()` method would indeed align better to this purpose.

Reply:
Thank you both. This has now been rewritten to a class with static members.